src/controller/retention/controller.go

```diff
@@ -41,31 +42,36 @@ type APIController interface {
 
 	DeleteRetention(id int64) error
```

Comment:
The first parameter of all functions in the interface should be context.
Please double check the usage of `orm.Context()`, this should be avoided